### PR TITLE
Updated & Fixed WebAPI

### PR DESCRIPTION
**Pull request by scotmcc**
Upgraded to .NET8.0

Fixed logging (removed Console replaced with Ilogger) 
Fixed warnings (Null Strings, etc).

The application now returns data from the back-end chat services when called from the Swagger API test framework.

### CODE REVIEW

**Comment by martindevans** [comment]
Thanks for this!

@xbotter I think you were working on some WebAPI stuff recently, would you mind reviewing this?

File: LLama.WebAPI/Models/SendMessageInput.cs [hunk 1]
```diff
@@ -2,15 +2,15 @@
...
 public class HistoryInput
 {
-    public List<HistoryItem> Messages { get; set; }
+    public List<HistoryItem>? Messages { get; set; }
     public class HistoryItem
     {
-        public string Role { get; set; }
-        public string Content { get; set; }
+        public string? Role { get; set; }
+        public string? Content { get; set; }
```
Comment (xbotter) [inline LLama.WebAPI/Models/SendMessageInput.cs:14]:
```suggestion
        public string Role { get; set; } = "User";
        public required string Content { get; set; }
```

File: LLama.WebAPI/Services/StatefulChatService.cs [hunk 2]
```diff
@@ -36,18 +38,15 @@ public void Dispose()
...
         }
-
-        Console.ForegroundColor = ConsoleColor.Green;
-        Console.Write(input.Text);
-
-        Console.ForegroundColor = ConsoleColor.White;
+        _logger.LogInformation("Input: {text}", input.Text);
         var outputs = _session.ChatAsync(
-            new Common.ChatHistory.Message(Common.AuthorRole.User, input.Text),
+            new Common.ChatHistory.Message(Common.AuthorRole.User, input.Text!),
```
Comment (xbotter) [inline LLama.WebAPI/Services/StatefulChatService.cs:49]:
```suggestion
            new Common.ChatHistory.Message(Common.AuthorRole.User, input.Text),
```

File: LLama.WebAPI/Controllers/ChatController.cs [hunk 3]
```diff
@@ -43,9 +43,9 @@ public async Task<string> SendHistory([FromBody] HistoryInput input, [FromServic
         {
             var history = new ChatHistory();
 
-            var messages = input.Messages.Select(m => new ChatHistory.Message(Enum.Parse<AuthorRole>(m.Role), m.Content));
+            var messages = input?.Messages?.Select(m => new ChatHistory.Message(Enum.Parse<AuthorRole>(m.Role!), m.Content!));
```
Comment (xbotter) [inline LLama.WebAPI/Controllers/ChatController.cs:46]:
```suggestion
            var messages = input.Messages.Select(m => new ChatHistory.Message(Enum.Parse<AuthorRole>(m.Role), m.Content));
```

File: LLama.WebAPI/Controllers/ChatController.cs [hunk 4]
```diff
@@ -43,9 +43,9 @@ public async Task<string> SendHistory([FromBody] HistoryInput input, [FromServic
...
 
-            history.Messages.AddRange(messages);
+            history.Messages.AddRange(messages!);
```
Comment (xbotter) [inline LLama.WebAPI/Controllers/ChatController.cs:48]:
```suggestion
            history.Messages.AddRange(messages);
```

**Comment by scotmcc** [comment]
Thanks for the feedback. I've added the default values you suggested.

**Review by xbotter — APPROVED** [review]
LGTM

**Pull request merged**